comp/trace/agent/def/component.go

```diff
@@ -6,7 +6,28 @@
 // Package traceagent provides the agent component type.
 package traceagent
 
+import (
+	"context"
+	"net/http"
+
+	pb "github.com/DataDog/datadog-agent/pkg/proto/pbgo/trace"
+
+	"github.com/DataDog/datadog-go/v5/statsd"
+	"github.com/DataDog/opentelemetry-mapping-go/pkg/otlp/attributes"
+	"github.com/DataDog/opentelemetry-mapping-go/pkg/otlp/attributes/source"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+)
+
 // team: agent-apm
 
 // Component is the agent component type.
-type Component interface{}
+type Component interface {
+	// SetStatsdClient sets the stats client of the underlying trace agent.
+	SetStatsdClient(mclient statsd.ClientInterface)
+	// SetOTelAttributeTranslator sets the OTel attributes translator of the underlying trace agent.
+	SetOTelAttributeTranslator(attrstrans *attributes.Translator)
+	// ReceiveOTLPSpans forwards the OTLP spans to the underlying trace agent to process.
+	ReceiveOTLPSpans(ctx context.Context, rspans ptrace.ResourceSpans, httpHeader http.Header) source.Source
```

Comment:
why do we need `httpHeader http.Header` ?  Whats its value for OTel Agent ?

Reply:
See below. This is required to pass the HTTP header `Datadog-Client-Computed-Stats` so that trace agent doesn't re-compute the APM stats for these traces.